agent/agent.go

```diff
@@ -1072,9 +1067,34 @@ LOAD:
 	return nil
 }
 
+// RegisterEndpoint registers a handler for the consul RPC server
+// under a unique name while making it accessible under the provided
+// name. This allows overwriting handlers for the golang net/rpc
+// service which does not allow this.
+func (a *Agent) RegisterEndpoint(name string, handler interface{}) error {
```

Comment:
As far as i can tell, this is only called from unit tests in the agent/ package, does it need to be exported like this or can we get away with renaming it to registerEndpoint now?